addon/components/select-component.js

```diff
@@ -307,17 +307,14 @@ export default Ember.Component.extend(
     } else {
       observableString = 'content.[]';
     }
+    let listItemFilter = this.get('_listItemFilter');
     ContentProxy = Ember.ObjectProxy.extend({
       _select: null,
       content: Ember.computed.alias('_select.content'),
       query: Ember.computed.alias('_select.query'),
       filteredContent: Ember.computed(function() {
-        var query, selectComponent;
-        selectComponent = this.get('_select');
-        query = this.get('query');
-        return (this.get('content') || []).filter(function(item) {
-          return selectComponent.matcher(query, item);
-        });
+        var query = this.get('query');
+        return (this.get('content') || []).filter(item => listItemFilter(query, item));
```

Comment:
```suggestion
        let content = this.get('content');
        return content ? content.filter(item => listItemFilter(query, item)) : [];
```

No need to call `filter` on an empty array.